File: python/hyperid/auth/discover.py
class Discover:
    def __init__(self, data):
        self.issuer = data.get('issuer')
        self.authorization_endpoint = data.get('authorization_endpoint')
        self.token_endpoint = data.get('token_endpoint')
        self.introspection_endpoint = data.get('introspection_endpoint')
        self.userinfo_endpoint = data.get('userinfo_endpoint')
        self.end_session_endpoint = data.get('end_session_endpoint')
        self.jwks_uri = data.get('jwks_uri')
        self.response_types_supported = data.get('response_types_supported', [])
        self.code_challenge_methods_supported = data.get('code_challenge_methods_supported', [])
        self.revocation_endpoint = data.get('revocation_endpoint')
        self.rest_api_token_endpoint = data.get('rest_api_token_endpoint')
        self.rest_api_public_endpoint = data.get('rest_api_public_endpoint')
        self.identity_providers = data.get('identity_providers', [])
        self.client_scopes_default = data.get('client_scopes_default', [])
        self.client_scopes_optional = data.get('client_scopes_optional', [])
        self.wallet_family = data.get('wallet_family', {})
        self.wallet_source = data.get('wallet_source', {})
        self.wallet_chain = data.get('wallet_chain', [])

    def get_scopes(self):
        combined_scopes = set(self.client_scopes_optional + self.client_scopes_default)
        return ' '.join(combined_scopes)

File: php/hyperid/base/discover.php
<?php

class Discover {
    public string $issuer;
    public string $authorizationEndpoint;
    public string $tokenEndpoint;
    public string $introspectionEndpoint;
    public string $userinfoEndpoint;
    public string $endSessionEndpoint;
    public string $jwksUri;
    public array $responseTypesSupported;
    public array $codeChallengeMethodsSupported;
    public string $revocationEndpoint;
    public string $restApiTokenEndpoint;
    public string $restApiPublicEndpoint;
    public array $identityProviders;
    public array $clientScopesDefault;
    public array $clientScopesOptional;
    public array $walletFamily;
    public array $walletSource;
    public array $walletChain;

    function __construct(array $data) {
        $this->issuer                           = $data['issuer'];
        $this->authorizationEndpoint            = $data['authorization_endpoint'];
        $this->tokenEndpoint                    = $data['token_endpoint'];
        $this->introspectionEndpoint            = $data['introspection_endpoint'];
        $this->userinfoEndpoint                 = $data['userinfo_endpoint'];
        $this->endSessionEndpoint               = $data['end_session_endpoint'];
        $this->jwksUri                          = $data['jwks_uri'];
        $this->responseTypesSupported           = $data['response_types_supported'];
        $this->codeChallengeMethodsSupported    = $data['code_challenge_methods_supported'];
        $this->revocationEndpoint               = $data['revocation_endpoint'];
        $this->restApiTokenEndpoint             = $data['rest_api_token_endpoint'];
        $this->restApiPublicEndpoint            = $data['rest_api_public_endpoint'];
        $this->identityProviders                = $data['identity_providers'];
        $this->clientScopesDefault              = $data['client_scopes_default'];
        $this->clientScopesOptional             = $data['client_scopes_optional'];
        $this->walletFamily                     = $data['wallet_family'];
        $this->walletSource                     = $data['wallet_source'];
        $this->walletChain                      = $data['wallet_chain'];
    }

    function getScopes() : string {
        $scopes = '';
        foreach ($this->clientScopesDefault as $scope){
            $scopes .= $scope.' ';
        }
        foreach ($this->clientScopesOptional as $scope){
            $scopes .= $scope.' ';
        }
        return trim($scopes);
    }
}

?>